Libraries/Opc.Ua.Server/Diagnostics/CustomNodeManager.cs

```diff
@@ -929,7 +917,7 @@ public virtual void DeleteAddressSpace()
         /// </remarks>
         public virtual object GetManagerHandle(NodeId nodeId)
         {
-            lock (Lock)
+            lock (m_lock)
```

Comment:
I'm not very deep into this, but there are about 10 places where a call to GetManagerHandle is protected by lock(Lock), one is protected by m_lock, and one (ore two?) are not protected by any lock. Which attribute is it we want to protect by the lock?

Reply:
in this case the read access to the NodeIdDictionary is protected. In a next improvement we could change the underlying dictionary to Concurrentdictionary, then the number of locks which cause contention can be greatly reduced. GetManagerHandle is the first candidate to have the lock removed. We tested without the lock but it could be unsafe if the node dictionary is modified. Lock and m_lock access the same lock here.